### PR TITLE
Add AdgenerationBidder and tests

### PR DESCRIPTION
**Pull request by AndriyPavlyuk**
Add AdgenerationBidder to keep up with Go version.

### CODE REVIEW

File: src/main/java/org/prebid/server/bidder/adgeneration/AdgenerationBidder.java [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+                    statusCode)));
+        }
+
+        final AdgenerationResponse adgenerationResponse;
+        try {
+            adgenerationResponse = decodeBodyToBidResponse(httpCall);
+        } catch (PreBidException e) {
+            return Result.emptyWithError(BidderError.badServerResponse(e.getMessage()));
+        }
+
```
Comment (BraslavskiyAndrey) [inline src/main/java/org/prebid/server/bidder/adgeneration/AdgenerationBidder.java:179]: missing 
if(CollectionUtils.isEmpty(adgenerationResponse.getResults())){
   return Result.empty();
}

Please look at adgeneration.go line 178

File: src/main/java/org/prebid/server/bidder/adgeneration/model/AdgenerationResponse.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    String locationid;
+
+    String dealid;
+
+    String ad;
+
+    String beacon;
+
+    String baconurl;
```
Comment (BraslavskiyAndrey) [inline src/main/java/org/prebid/server/bidder/adgeneration/model/AdgenerationResponse.java:20]: Type beaconUrl

File: src/main/java/org/prebid/server/bidder/adgeneration/AdgenerationBidder.java [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+        if (site != null && StringUtils.isNotBlank(site.getPage())) {
+            uriBuilder.addParameter("tp", site.getPage());
+        }
+
+        return uriBuilder.toString();
+    }
+
+    private String getAdSize(Imp imp) {
```
[the remark this answers is not in the document]
Comment (AndriyPavlyuk) [reply src/main/java/org/prebid/server/bidder/adgeneration/AdgenerationBidder.java:116]: Thanks! Pretty nice!

File: src/main/java/org/prebid/server/bidder/adgeneration/AdgenerationBidder.java [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+        }
+
+        if (adSize.length() > 0 && adSize.lastIndexOf(",") == adSize.length() - 1) {
+            adSize.deleteCharAt(adSize.length() - 1);
+        }
+
+        return adSize.toString();
+    }
+
+    private String getCurrency(BidRequest request) {
```
[the remark this answers is not in the document]
Comment (AndriyPavlyuk) [reply src/main/java/org/prebid/server/bidder/adgeneration/AdgenerationBidder.java:134]: Thanks, looks pretty nice and shorter!

File: src/main/java/org/prebid/server/bidder/adgeneration/model/AdgenerationResponse.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
+package org.prebid.server.bidder.adgeneration.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Value;
+
+import java.math.BigDecimal;
+
+@AllArgsConstructor(staticName = "of")
+@Value
+public class AdgenerationResponse<T> {
```
[the remark this answers is not in the document]
Comment (AndriyPavlyuk) [reply src/main/java/org/prebid/server/bidder/adgeneration/model/AdgenerationResponse.java:10]: Ok, i removed generic from class and added check for results empty.

baconurl, ttl, landingUrl, scheduleid are used in Go tests. They include this parameters in   AdgenerationResponse.results. So maybe we need this fields for correct flow?

File: src/main/java/org/prebid/server/bidder/adgeneration/AdgenerationBidder.java [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+            return mapper.decodeValue(httpCall.getResponse().getBody(), AdgenerationResponse.class);
+        } catch (DecodeException e) {
+            throw new PreBidException(e.getMessage(), e);
+        }
+    }
+
+    private String getAdm(AdgenerationResponse adgenerationResponse, String impId) {
+        String ad = adgenerationResponse.getAd();
+        if (StringUtils.isNotBlank(adgenerationResponse.getVastxml())) {
+            ad = "<body><div id=\"apvad-" + impId + "\"></div><script type=\"text/javascript\" id=\"apv\" "
```
[the remark this answers is not in the document]
Comment (AndriyPavlyuk) [reply src/main/java/org/prebid/server/bidder/adgeneration/AdgenerationBidder.java:219]: Yes, thanks

**Comment by DGarbar** [comment]
In GO they expects currency "JPA" but it is "USD" that really sends in bid. (They just do not check that parameter in tests)

**Pull request merged**